javascript-injectors/shiphero-signin-via-credentials.js

```diff
@@ -0,0 +1,32 @@
+(function () {
+  const email = '<YOUR_EMAIL>'
+  const password = '<YOUR_PASSWORD>'
+
+  function changeValue (input, value) {
+    const nativeInputSetter = Object.getOwnPropertyDescriptor(
+      window.HTMLInputElement.prototype,
+      'value'
+    ).set
+
+    nativeInputSetter.call(input, value)
+
+    const inputEvent = new Event('input', { bubbles: true })
+    input.dispatchEvent(inputEvent)
+  }
+
+  function login () {
+    try {
+      const emailEl = document.querySelector('input[name="email"]')
+      const passwordEl = document.querySelector('input[name="password"]')
+
+      changeValue(emailEl, email)
+      changeValue(passwordEl, password)
+
+      document.querySelector('button[name="submit"]').click()
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  setTimeout(login, 5000)
```

Comment:
```suggestion
(function () {
  const email = '<YOUR_EMAIL>'
  const password = '<YOUR_PASSWORD>'
  const authLocation = '/some/signin/page'
  
  function setValue (selector, value) {
    const element = document.querySelector(selector)
    element.value = value;
    element.dispatchEvent(new Event('change'));
  }
  
  function submitForm () {
      document.querySelector('button[name="submit"]').click()
  }
  
  function login () {
    try {
      setValue('input[name="email"]', email)
      setValue('input[name="password"]'', password)
      submitForm()
    } catch (error) {
      console.warn(error)
      setTimeout(login, 1000)
    }
  }

  if (window.location.pathname === authLocation) {
      login()
  }
```

Reply:
@rusko124 

This snippet does not work for Ship Hero. It seems to be React based, so normal events don't seem to work.

`const element = document.querySelector(selector)
    element.value = value;
    element.dispatchEvent(new Event('change'));`

Reply:
This is kinda weird to me but OK, let's use the recommendation.

Reply:
Updated to fix pending lint issues